dmutils/terms_of_use.py

```diff
@@ -0,0 +1,28 @@
+from flask import session, redirect, request
+
+# A user who hasn't agreed to the most recent version of the terms of use needs to be redirected to a page to review and
+# agree to the latest terms.
+#
+# To minimise disruption, the staleness of the terms agreement is only checked at certain key points in the site, such
+# as at login.  Then a flag is set in the session to trigger redirection to the terms review page.
+
+
+NEEDS_UPDATE_SESSION_VAR = 'terms_update'
+REVIEW_URL = '/terms-updated'
+
+
+def init_app(application):
+    static_path = application.static_url_path
+
+    @application.before_request
+    def redirect_for_update():
+        if needs_acceptance_update() and request.path != REVIEW_URL and not request.path.startswith(static_path):
```

Comment:
Another way you could have done this is create a decorator and then explicitly mark the views that need to check acceptance